### ISSUE TITLE
E101 False positive in multiline string

### ISSUE THREAD

**Issue by david-waterworth** [astral-sh/ruff#7906]
I have a python model that contains rest endpoint queries defined as multi-line strings, i.e. 

```
query = """
    query ...
"""
```

`ruff` is reporting `E101 Indentation contains mixed spaces and tabs` for the text inside the string.

Technically this is correct as the indentation of the query does contain a mix of tab and spaces - but in general, I feel that `ruff` shouldn't try and lint a text block.

Note I observed this in a notebook - I've not tested in a py file

**Comment by charliermarsh** [comment]
So pycodestyle has this behavior too, though it's unclear whether they consider it correct (https://github.com/PyCQA/pycodestyle/issues/600, https://github.com/PyCQA/pycodestyle/issues/376, https://github.com/PyCQA/pycodestyle/issues/45).

We actually disabled W109 (which flags uses of tabs) within such strings. I'm open to disabling E101 too -- but what's the valid use-case for mixing spaces and tabs within a single indent, even within a string?

**Comment by njzjz** [comment]
In a unit test, one may want to check if a file has the expected content.

```py
a = """
	    example1
	    example2
"""
with open("some_file.txt") as f:
    b = f.read()
assert a == b
```

The fix breaks the test.

**Comment by MichaReiser** [comment]
Ignoring leading whitespace in multiline strings seems reasonable because it is not an indentation.
